### PR TITLE
Add ADR 0003 medium for router configurations storage

### PR DESCRIPTION
**Pull request by danialmalik**
**Description:** 
Add ADR 0003 that describes the medium we will be using for storing the router configurations.

**JIRA:** N/A

**Dependencies:** [event-tracking](https://github.com/edx/event-tracking) 

**Merge deadline:** N/A

**Installation instructions:** N/A

**Testing instructions:** N/A

**Reviewers:**
- [ ] tag reviewer 
- [ ] tag reviewer 

**Merge checklist:**
- [ ] All reviewers approved
- [ ] CI build is green
- [ ] Version bumped
- [ ] Changelog record added
- [ ] Documentation updated (not only docstrings)
- [ ] Commits are squashed

**Post merge:**
- [ ] Create a tag
- [ ] Check new version is pushed to PyPI after tag-triggered build is 
      finished.
- [ ] Delete working branch (if not needed anymore)

**Author concerns:** N/A

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @danialmalik! I've created [OSPR-4999](https://openedx.atlassian.net/browse/OSPR-4999) to keep track of it in JIRA, where we prioritize reviews. Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).

Please let us know once your PR is ready for our review and all tests are green.

File: docs/decisions/0003-router-configurations-storage.rst [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+-  Store configurations in Django settings (YAML configuration files).
+
+-  Store configurations in some Django model (database).
+
+Decision
+--------
+
+Keeping in mind a large number of enterprise clients, we will store
+router configurations in `Django configuration models
+<https://github.com/edx/django-config-models>`__ to keep things simple.
```
Comment (feanil) [inline docs/decisions/0003-router-configurations-storage.rst:28]:
```suggestion
<https://github.com/edx/django-config-models>`__.
```

File: docs/decisions/0003-router-configurations-storage.rst [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+Consequences
+------------
+
+Storing configurations in `Django configuration
+models <https://github.com/edx/django-config-models>`__ will keep the
+settings files simple and will avoid unnecessary complexities.
```
Comment (feanil) [inline docs/decisions/0003-router-configurations-storage.rst:35]:
```suggestion
settings files simple.
```
That language is a bit vague.

File: docs/decisions/0003-router-configurations-storage.rst [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+Django configuration models have built-in caching support which would
+help us address performance-related challenges.
+
```
Comment (feanil) [inline docs/decisions/0003-router-configurations-storage.rst:39]: Another reason to add here is that enterprise customer management might be done by teams other than the operators and putting it in the database here allows us to let customer support teams manage the settings or potentially make it self service in the future.

**Review by feanil — CHANGES_REQUESTED** [review]

File: docs/decisions/0003-router-configurations-storage.rst (hunk 3 above)
Comment (danialmalik) [reply docs/decisions/0003-router-configurations-storage.rst:39]: Updated.

**Review by feanil — APPROVED** [review]

**Comment by edx-webhook** [comment]
@danialmalik 🎉 Your pull request was merged!

Please take a moment to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=edx/event-routing-backends&entry.867055334=https://github.com/edx/event-routing-backends/pull/3&entry.1484655318=https://github.com/danialmalik&entry.752974735=2020-09-23+15:50&entry.1917517419=2020-09-24+19:04&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**